ir/memory.cpp

```diff
@@ -195,19 +211,17 @@ expr Pointer::get_offset() const {
   return p.extract(m.bitsOffset() - 1, 0).sextOrTrunc(m.bits_size_t);
 }
 
-expr Pointer::get_value(const char *name, const FunctionExpr &fn,
-                        const expr &ret_type) const {
-  auto bid = get_short_bid();
-  return expr::mkIf(is_local(), fn(bid), expr::mkUF(name, { bid }, ret_type));
-}
-
-expr Pointer::get_address(bool simplify) const {
+expr Pointer::get_address(bool simplify,
+                          optional<function<expr(const expr&)>> local_addrs)
+                          const {
```

Comment:
`local_addrs` is given when `m.local_blk_addr` cannot be used due to the existence of quantifier.
The main motivation here was that, a declaration of recursive function cannot be nested inside forall expression, so this returns a wrong expression if `m.local_blk_addr` contains quantifiers.